pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/MetadataResponseFormatter.java

```diff
@@ -55,50 +54,65 @@ public static String formatResponseString(Metadata metadata) throws IOException
      * To be used as the result string for HAWQ.
      * An example result is as follows:
      *
-     * {"PXFMetadata":[{"table":{"dbName":"default","tableName":"t1"},"fields":[{"name":"a","type":"int"},{"name":"b","type":"float"}]}]}
+     * {"PXFMetadata":[{"item":{"path":"default","name":"t1"},"fields":[{"name":"a","type":"int"},{"name":"b","type":"float"}]}]}
      */
-    private static String metadataToJSON(Metadata metadata) throws IOException {
+    private static String metadataToJSON(List<Metadata> metadataList) throws IOException {
```

Comment:
Have a look at the way we serialize the Fragments - we had a problem of having too many fragments, so we needed to serialize them inside a streaming object. 
I think that now that the metadata can include multiple tables, the same approach should be taken here as well.
The same applies also for the debug function that prints all of the items - if there are too many of them the StringBuilder will run out of memory. The solution in the fragments case was to print a log of one fragment at a time - perhaps the same should be done here as well.